### DISCUSSION TITLE
Solana sign error

### DISCUSSION THREAD

**Discussion by Zeckie99** [community/community#133666] [Programming Help]
### Select Topic Area

Question

### Body

Hi,

I'm  beginner in programming and I'm developing a PY project on the Solana. I need to perform a swap operation to buy a specific token, but I'm encountering an error. Here is the error message I'm getting:

Creating swap instructions...
amm_id: GUgCKwn5SiBudJLaMXYa15m8xB4iH6EPwWJB2by6Dbo6
amm_authority: 675kPX9MHTjS2zt1qfr1NYHuzeLXfQM9H24wFSUt1Mp8
pool_coin_token_account: 8nfhMGe9cCkBV36NtNriDJa31T8MPd6HSPsE65zBKiFx
pool_pc_token_account: So11111111111111111111111111111111111111112
Swap instructions created
Close account params created
Recent blockhash: DmbPcxoecDT89PRdbd7RbWUvHbdJrT4DrNmRMbsGXBjV
Unhandled exception: argument 'keypairs': failed to extract enum Signer ('KeypairWrapper | PresignerWrapper | NullSignerWrapper')
- variant KeypairWrapper (KeypairWrapper): TypeError: failed to extract field Signer::KeypairWrapper.0, caused by TypeError: 'list' object cannot be converted to 'Keypair'
- variant PresignerWrapper (PresignerWrapper): TypeError: failed to extract field Signer::PresignerWrapper.0, caused by TypeError: 'list' object cannot be converted to 'Presigner'
- variant NullSignerWrapper (NullSignerWrapper): TypeError: failed to extract field Signer::NullSignerWrapper.0, caused by TypeError: 'list' object cannot be converted to 'NullSigner'. Retrying...

Here is the relevant part of my code:

tx = Transaction(fee_payer=keypair.pubkey(), recent_blockhash=recent_blockhash)
tx.add(instructions_swap)
tx.add(set_compute_unit_price(compute_unit_price))
tx.add(set_compute_unit_limit(compute_unit_limit))
tx.add(closeAcc)

tx.sign([keypair, wsol_account_keypair])  # I'm getting an error on this line

# Serialize the transaction
serialized_tx = tx.serialize()
txn = solana_client.send_raw_transaction(serialized_tx)
txid_string_sig = txn['result']

Error;
Unhandled exception: argument 'keypairs': failed to extract enum Signer ('KeypairWrapper | PresignerWrapper | NullSignerWrapper')        
- variant KeypairWrapper (KeypairWrapper): TypeError: failed to extract field Signer::KeypairWrapper.0, caused by TypeError: 'list' object cannot be converted to 'Keypair'
- variant PresignerWrapper (PresignerWrapper): TypeError: failed to extract field Signer::PresignerWrapper.0, caused by TypeError: 'list' object cannot be converted to 'Presigner'
- variant NullSignerWrapper (NullSignerWrapper): TypeError: failed to extract field Signer::NullSignerWrapper.0, caused by TypeError: 'list' object cannot be converted to 'NullSigner'. Retrying...

It looks like tx.sign([keypair, wsol_account_keypair]) expects arguments of types KeypairWrapper, PresignerWrapper, or NullSignerWrapper, but I might be doing something wrong. I'm not sure if keypair and wsol_account_keypair are of the correct types. I need some help with this.

Any assistance would be greatly appreciated, thank you

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @Zeckie99 ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
